continuum/scenarios/transformation_incremental.py

```diff
@@ -0,0 +1,68 @@
+from typing import Callable, List
+
+import numpy as np
+
+from continuum.task_set import TaskSet
+from continuum.datasets import _ContinuumDataset
+from continuum.scenarios import InstanceIncremental
+
+from torchvision import transforms
+
+
+class TransformationIncremental(InstanceIncremental):
+    """Continual Loader, generating datasets for the consecutive tasks.
+
+    Scenario: Every task contains the same data with different transformation
+    It is a cheap way to create instance incremental scenarios.
+    Classic transformation incremental scenarios are "permutations" and "rotations"
+
+    :param cl_dataset: A continual dataset.
+    :param nb_tasks: Number of tasks in the continuum.
+    :param incremental_transformations: list of transformations to apply to specific tasks
+    :param base_transformations: List of transformation to apply to all tasks.
+    """
+
+    def __init__(
+            self,
+            cl_dataset: _ContinuumDataset,
+            nb_tasks: int,
+            incremental_transformations: List[List[Callable]],
+            base_transformations: List[Callable] = None
+    ):
+        super(TransformationIncremental, self).__init__(
+            cl_dataset=cl_dataset,
+            nb_tasks=nb_tasks,
+            base_transformations=base_transformations
+        )
+        if incremental_transformations is None:
+            raise ValueError("For this scenario a list transformation should be set")
+
+        if nb_tasks != len(incremental_transformations):
+            raise ValueError("The number of tasks is not equal to the number of transformation")
+
+        self.inc_trsf = incremental_transformations
+        self._nb_tasks = self._setup(nb_tasks)
+
+    def get_task_index(self, nb_tasks, y):
+        # all tasks have all labels, only the transformation change
+        return y
+
+    def get_task_transformation(self, task_index):
+        return transforms.Compose(self.inc_trsf[task_index] + self.trsf.transforms)
+
+    def update_task_indexes(self, task_index):
+        new_t_ = np.ones(len(self.dataset[1])) * task_index
+        self.dataset = (self.dataset[0], self.dataset[1], new_t_)
+
+    def __getitem__(self, task_index):
```

Comment:
The other scenarios and the abstract class accept either a task index (int) or several task index (slice). The latter case is not supported here (and I think it's correct), but you should have a check if/except if `task_index` is a `slice` to avoid any misunderstanding.

Reply:
Can you do it and push on patch_pr29 branch ?

Reply:
Done.